src/dbnode/storage/shard.go

```diff
@@ -1503,30 +1510,44 @@ func (s *dbShard) insertSeriesBatch(inserts []dbShardInsert) error {
 			// this method (insertSeriesBatch) via `entryRefCountIncremented` mechanism.
 			entry.OnIndexPrepare()
 
-			id := entry.Series.ID()
-			tags := entry.Series.Tags().Values()
-
-			var d doc.Document
-			d.ID = id.Bytes() // IDs from shard entries are always set NoFinalize
-			d.Fields = make(doc.Fields, 0, len(tags))
-			for _, tag := range tags {
-				d.Fields = append(d.Fields, doc.Field{
-					Name:  tag.Name.Bytes(),  // Tags from shard entries are always set NoFinalize
-					Value: tag.Value.Bytes(), // Tags from shard entries are always set NoFinalize
-				})
+			// Don't insert cold index writes into the index insert queue.
+			if s.coldWritesEnabled &&
+				writeType == series.ColdWrite &&
+				pendingIndex.timestamp.Before(warmIndexBlockStart) {
+				indexBlockStart := s.reverseIndex.BlockStartForWriteTime(pendingIndex.timestamp)
+				// NB(bodu): We to mark this entry as indexed for `indexBlockStart`. This means that
+				// we will not attempt to index this series for this time block again.
+				entry.OnIndexSuccess(indexBlockStart)
```

Comment:
I think we also need `entry.OnIndexFinalize(indexBlockStart)` too to decrement the entry count.

For reference this happens once we index it.
https://github.com/m3db/m3/blob/master/src/dbnode/storage/index/types.go#L662-L663
```go
			b.entries[idx].OnIndexSeries.OnIndexSuccess(blockStart)
			b.entries[idx].OnIndexSeries.OnIndexFinalize(blockStart)
```